backend/parser.py

```diff
@@ -49,42 +68,100 @@ def add_point(self, timestamp, data: list, latency=None):
         else:
             self.timestamp_diffs.append(0)
 
-        if latency != None:
+        if latency >= 0:
             self.latencies.append(latency)
 
-        for i, point in enumerate(data):
+        for i, point in enumerate(datapoints):
             self.data[i].append(point)
             
         self.next_index += 1
 
+    def set_dimensions(self, dimensions: int):
+        """Initializes empty lists in dict after parser determines data dimensions
+        
+        Args:
+            dimensions: The number of dimensions that the data has. e.g. if the
+                gyroscope returns (x,y,z) tuples the parser calls set_dimensions(3)
+        
+        Raises:
+            ValueError: "<= 1 dimension error" 1 dimensions is always initialized and negative/0 dimensions don't exist
+        """
+
+        if dimensions <= 1:
+            raise ValueError("<= 1 dimension error")
```

Comment:
I think if someone passes '1', you can return without changing anything or throwing an error.